pkg/kubeadm/kubeadm.go

```diff
@@ -162,6 +163,11 @@ func (k *Config) CreatePKI() (err error) {
 
 // CreateKubeConfig - Creates all the kubeconfig files requires for masters
 func (k *Config) CreateKubeConfig() (err error) {
+	if k.KubeletID == "" {
+		if k.KubeletID, err = os.Hostname(); err != nil {
```

Comment:
The container (keto-k8) runs with --net=host to properly support kubeadm for this and a hoot of other things (validation checks, systemd intergration etc)...